Sources/Operators/ShareReplay.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  ShareReplay.swift
+//  CombineExt
+//
+//  Created by Jasdev Singh on 13/04/2020.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
+
+import Combine
+
+public extension Publisher {
+    /// An overload on [Publisher.share](https://developer.apple.com/documentation/combine/publisher/3204754-share)
+    /// that allows for buffering and replaying a `replay` amount of value events to future subscribers.
+    /// - Parameter count: The number of value events to buffer in a [first-in-first-out](https://en.wikipedia.org/wiki/FIFO_(computing_and_electronics)) manner.
+    /// - Returns: A type-erased, shared publisher that replays the specified number of value events to future subscribers.
+    func share(replay count: UInt) -> AnyPublisher<Output, Failure> {
+        multicast { ReplaySubject(maxBufferSize: count) }
```

Comment:
Might be worth specializing for 1 with CurrentValueSubject and over 1 with ReplaySubject ? 

```suggestion
        multicast { ReplaySubject(maxBufferSize: count) }
```

Reply:
Also, could be cool to return `Publishers.Multicast` instead of erasing the entire thing (while erasing the inner-constraints of the multi-cast, perhaps)

Reply:
> Might be worth specializing for 1 with CurrentValueSubject

Hah! So part of my next blog post digs into this—it’s subtle. But, we can’t lean on `CurrentValueSubject` since it behaves like a `BehaviorSubject` and doesn’t replay any values post-terminal event.

And game to return a `Publishers.Multicast` with an erased inner. Incoming in the next commit.

Reply:
That's interesting :) Well, we have BehaviorRelay in the repo which would help with that, but sure we can leave it as is :] thanks!